nixos/modules/services/games/asf.nix

```diff
@@ -0,0 +1,200 @@
+    };
+
+    config = mkOption {
+      type = types.attrs;
+      description = ''
+        The ASF.json file, all the options are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#global-config.
+        Do note that AutoRestart is automatically set to false, since nix takes care of updating everything.
+        UpdateChannel is also set to 0 for the same reason.
+        Headless is also set to true, since there is no way to provide input via a systemd service.
+        You should try to keep ASF up to date since upstream does not provide support for anything but hte latest version and you're exposing yourself to all kinds of issues - as is outlined here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#updateperiod
+      '';
+      example = {
+        Statistics = false;
+      };
+      default = { };
+    };
+    bots = mkOption {
+      type = types.attrsOf (types.submodule {
+        options = {
+          username = mkOption {
+            type = types.str;
+            description =
+              "Name of the user to log in. Default is attribute name.";
+            default = "";
+          };
+          passwordFile = mkOption {
+            type = types.path;
+            description =
+              "Path to a file containig the password. The file must be readable by the <literal>asf</literal> user/group.";
+          };
+          enabled = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether to enable the bot on startup.";
+          };
+          settings = mkOption {
+            type = types.attrs;
+            description =
+              "Additional settings that are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#bot-config";
+            default = { };
+          };
+        };
+      });
+      description = ''
+        Bots name and configuration
+      '';
+      example = {
+        exampleBot = {
+          username = "alice";
+          passwordFile = "/my/super/secret/password";
+          settings = { SteamParentalCode = "1234"; };
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users = {
+      users.asf = {
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+        group = "asf";
+        description = "Archis-Steam-Farm service user";
+      };
+      groups.asf = { };
+    };
+
+    systemd.services = {
+      asf = {
+        description = "Archis-Steam-Farm Service";
+        after = [ "network.target" ];
+
+        serviceConfig = {
+          User = "asf";
+          Group = "asf";
+          WorkingDirectory = cfg.dataDir;
+          Type = "simple";
+          ExecStart =
+            "${cfg.package}/bin/ArchiSteamFarm --path ${cfg.dataDir} --process-required --no-restart --service ";
+
+          # mostly copied from the default systemd service
```

Comment:
Can we reuse parts of the upstream service to avoid falling out of sync with the upstream service? Falling out of sync with the upstream service is [something that happens from time to time in NixOS](https://github.com/NixOS/nixpkgs/pull/147305).

Reply:
Not all security hardenings from the upstream service template can be used here, upstream makes a few assumptions.

Reply:
This comment was only valid if we used `systemd` templates and you explained we aren't so this comment can be disregarded anyways. Thanks for explaining!